src/kibali/AuthZChecker.cs

```diff
@@ -82,6 +86,34 @@ public AccessRequestResult CanAccess(string url, string method, string scheme, s
             return AccessRequestResult.InsufficientPermissions;
         }
 
+        public void Validate(PermissionsDocument permissionsDocument)
+        {
+            // Walk permissions, find each pathSet and add path to dictionary
+            foreach (var permission in permissionsDocument.Permissions)
+            {
+                foreach(var pathSet in permission.Value.PathSets)
+                {
+                    foreach (var path in pathSet.Paths)
+                    {
+                        ProtectedResource resource;
+                        if (resources.ContainsKey(path.Key))
+                        {
+                            resource = resources[path.Key];
+
+                        }
+                        else
+                        {
+                            resource = new ProtectedResource(path.Key);
+                            resources.Add(path.Key, resource);
+                        }
+                        resource.ValidateLeastPrivilegePermissions(permission.Key, pathSet, path.Value.LeastPrivilegedPermission);
+                        this.ContainsErrors |= resource.ContainsErrors;
+                        this.Errors.UnionWith(resource.PermissionsErrors);
```

Comment:
If we do stick with this design that holds onto Errors then we should clear before we start validating or we will never be able to get rid of errors without throwing away the AuthZChecker.

Reply:
Updated to return a collection of errors.